rt/rs/description-swagger/src/main/java/org/apache/cxf/jaxrs/swagger/SwaggerUiResolver.java

```diff
@@ -87,12 +74,22 @@ protected static String checkUiRoot(String urlStr, String swaggerUiVersion) {
     }
 
     public static String findSwaggerUiRoot(String swaggerUiMavenGroupAndArtifact, 
-                                           String swaggerUiVersion) {
-        String root = HELPER.findSwaggerUiRootInternal(swaggerUiMavenGroupAndArtifact, 
-                                                       swaggerUiVersion);
-        if (root == null && HELPER.getClass() != SwaggerUiResolver.class) {
-            root = new SwaggerUiResolver().findSwaggerUiRootInternal(swaggerUiMavenGroupAndArtifact, 
-                                                                     swaggerUiVersion);
+                                           String swaggerUiName, String swaggerUiVersion) {
+        String root = null;
+        try {
+            root = new OsgiSwaggerUiResolver()
+                .findSwaggerUiRoot(swaggerUiMavenGroupAndArtifact, swaggerUiVersion);
+        } catch (Throwable e) {
+            // Ignore
+        }
+        try {
```

Comment:
Should it be wrapped into `if (root == null) { ... }` block (if previous lookup succeeded)?